backend/tests/api/announcement/index.js

```diff
@@ -0,0 +1,200 @@
+const request = require('supertest');
+const { jwt } = require("../../helpers/strapi");
+const userFactory = require("../../user/factory");
+
+describe("Announcement Test", () => {
+    let authenticatedUser;
+    let adminUser;
+
+    beforeAll(async () => {
+        authenticatedUser = await userFactory.createUser(strapi, "authenticated");
+        adminUser = await userFactory.createUser(strapi, "admin");
+    })
+
+    const constructAnnouncement = (id) => {
+        return {
+            data: {
+                "Title": "Test Announcement " + id,
+                "Poster": null,
+                "Description": "Announcement Description " + id,
+                // "HostingGroup": "Announcement HostingGroup" + id,
+                "Private": false,
+                "locale": "en",
+                "publishedAt": Date.now(),
+            }
+        };
+    }
+
+    // Create
+    it("[Create] Public user should not post Announcements", async () => {
+        await request(strapi.server.httpServer)
+            .post("/api/announcements")
+            .set("accept", "application/json")
+            .send(constructAnnouncement(0))
+            .expect("Content-Type", /json/)
+            .expect(500)
+    })
+
+    it("[Create] Authenticated user should not post Announcements", async () => {
+        await request(strapi.server.httpServer)
+            .post("/api/announcements")
+            .set("accept", "application/json")
+            .set("Authorization", `Bearer ${await jwt(authenticatedUser.id)}`)
+            .send(constructAnnouncement(0))
+            .expect("Content-Type", /json/)
+            .expect(500)
+    })
+
+    it("[Create] Admin user should post Announcements", async () => {
+        const announcement = constructAnnouncement(1);
+        await request(strapi.server.httpServer)
+            .post("/api/announcements")
+            .set("accept", "application/json")
+            .set("Authorization", `Bearer ${await jwt(adminUser.id)}`)
+            .send(constructAnnouncement(1))
+            .expect("Content-Type", /json/)
+            .expect(200)
+            .expect((data) => {
+                data = data.body.data;
+                expect(data.attributes.Title).toBe(announcement.data.Title);
+                // expected "null", but received "undefined"
```

Comment:
Is this comment needed? If this is a question for us, make a review comment. If it's an unnecessary comment, it should be deleted.

Reply:
okay sounds good.